src/lang/base/JSON.ml

```diff
@@ -65,12 +65,13 @@ let literal_type_exn l =
     pp_typ s
 
 let build_prim_lit_exn t v =
-  let exn_wrapper t v r = 
-    match v with
-    | None -> raise (mk_invalid_json ("Invalid " ^ (pp_typ t) ^ " value " ^ r ^ " in JSON"))
-    | Some v' -> v'
-  in
-    exn_wrapper t (build_prim_literal t v) v
+  let exn = mk_invalid_json ("Invalid " ^ (pp_typ t) ^ " value " ^ v ^ " in JSON") in
```

Comment:
The pattern used here is to create the error message before checking whether an error has occurred. Didn't we have a problem with this pattern some time ago, @vaivaswatha?

I think the issue was when printing values rather than types, but in principle the type could also be quite big, so the call to `pp_typ` could be costly.

Reply:
Good catch @jjcnn! I fixed this one and a less expensive string computation elsewhere.

Reply:
> The pattern used here is to create the error message before checking whether an error has occurred. Didn't we have a problem with this pattern some time ago, @vaivaswatha?
> 

You're right, I missed this. That was for `literal`s for which the problem was more critical, but good to change this too.